app/src/main/java/de/dhbw/corona_world_app/datastructure/displayables/GermanyState.java

```diff
@@ -0,0 +1,51 @@
+package de.dhbw.corona_world_app.datastructure.displayables;
+
+import de.dhbw.corona_world_app.datastructure.Displayable;
+
+public enum GermanyState implements Displayable {
+    BADEN_WUERTTEMBERG("DE-BW","Baden-Wuerttemberg"),
+    BAYERN("DE-BY","Bayern"),
+    BERLIN("DE-BE","Berlin"),
+    BRANDENBURG("DE-BB","Brandenburg"),
+    BREMEN("DE-HB","Bremen"),
+    HAMBURG("DE-HH","Hamburg"),
+    HESSEN("DE-HE","Hessen"),
+    MECKLENBURG_VORPOMMERN("DE-MV","Mecklenburg-Vorpommern"),
+    NIEDERSACHSEN("DE-NI","Niedersachsen"),
+    NORDRHEIN_WESTFALEN("DE-NW","Nordrhein-Westfalen"),
+    RHEINLAND_PFALZ("DE-RP","Rheinland-Pfalz"),
+    SAARLAND("DE-SL","Saarland"),
+    SACHSEN("DE-SN","Sachsen"),
+    SACHSEN_ANHALT("DE-ST","Sachsen-Anhalt"),
+    SCHLESWIG_HOLSTEIN("DE-SH","Schleswig-Holstein"),
+    THUERINGEN("DE-TH","Thueringen"),
+    ;
```

Comment:
I would have made this extend ISOCountry instead of making a whole new interface and casting every country name to ISOCountry. Your approach may cause problems in the future. if it isnt too much work consider my approach

Reply:
enums cant extend they can only implement an interface

Reply:
I chose this approach so that there are no problems in the future (if we were thinking about adding other countries). All the problems I did have now with the merge and as I did not cut anything away from the functionality of isoCountry I don't see where this would cause any problems in the forseeable future.

Reply:
Yes i have forgotten that you cannot extend enums. The biggest Problem that i have is the casting into ISOCountry. If you were to implement many Classes of this interface it could lead to a lot of unnecassery instanceof checks. My solution for this Problem would be to let Country implement a generic that extends Displayable (we should probably change that name) so that it is known what type of displayable is used in the Country Object.

Reply:
Hmm. Why would you cast it to ISOCountry? As is, all the functionalities are equal between Displayable, ISOCountry and GermanyState...

Reply:
You are casting the country.getName to IsoCountry in your code multiple times (Apimanager line 69, MapFragment line 298). More Classes that implement displayable will be added when more countries are added, this could cause a lot of issues. When you use a generic you always know the Type of the Displayable that is used in Country, with your approach it is not guaranteed to work. Use a Generic in Country so we implicitly have the right type without needing to cast it to the type, which could cause unnecessary problems.

Reply:
Yes, that is because all called methods in your examples expect ISOCountry and not Displayable... I don't quite understand what is supposed to be dangerous about these casts. These are necessary if you don't want to refactor all the methods and replace all occurences of ISOCountry with Displayable (which btw would actually be profitable). But I did not want to refactor your or my code out of a completely different branch.

Reply:
After disscussion, a generic will be implemented to ensure type safety

Reply:
I have quite a lot of trouble with that approach... (every instance of country needs to be changed)

Reply:
Ok generics now work, I will fix everything and then push. Pull request should then be ready to go!

Reply:
Push is out, resolving conversation